packages/backend/src/open_payments/receiver/service.ts

```diff
@@ -324,12 +334,27 @@ async function getIncomingPaymentGrant(
   )
 
   if (isNonInteractiveGrant(grant)) {
-    return await deps.grantService.create({
-      ...grantOptions,
-      accessToken: grant.access_token.value,
-      expiresIn: grant.access_token.expires_in
-    })
+    try {
+      return await deps.grantService.create({
+        ...grantOptions,
+        accessToken: grant.access_token.value,
+        managementId: retrieveManagementId(grant.access_token.manage),
+        expiresIn: grant.access_token.expires_in
+      })
+    } catch (err) {
+      deps.logger.warn({ grantOptions }, 'Grant has wrong format')
+      return undefined
+    }
   }
   deps.logger.warn({ grantOptions }, 'Grant request required interaction')
   return undefined
 }
+
+function retrieveManagementId(managementUrl: string): string {
+  const managementUrlParts = managementUrl.split('/')
+  const managementId = managementUrlParts.pop() || managementUrlParts.pop() // handle trailing slash
+  if (!managementId) {
+    throw new Error('invalid management id')
+  }
```

Comment:
If we did response checking of the management url in the client, I wouldn't have to do that.